File: shared-datacenter-proxies/GoLang/proxy.go
package main

import (
	"fmt"
	"strings"
)

func createProxyByUrl(proxyMap map[string]string, url string) (string, string) {
	urlParts := strings.Split(url, ";")
	proxyAddress, _ := proxyMap[DefaultProxyIndexName]

	if len(urlParts) == 2 {
		url = urlParts[0]
		country := urlParts[1]

		if countrySpecificProxy, ok := proxyMap[country]; ok {
			proxyAddress = countrySpecificProxy
		}
	}

	return url, fmt.Sprintf(
		"http://customer-%s:%s@%s",
		Username,
		Password,
		proxyAddress,
	)
}

File: shared-datacenter-proxies/Ruby/proxy.rb
require_relative './settings'

def create_proxy_by_url(proxy_map, url)
  url_parts = url.split ';'
  default_proxy_index = get_setting :DefaultProxyIndexName

  country = nil
  if url_parts.length == 2
    url = url_parts[0]
    country = url_parts[1]
  end

  proxy_address = proxy_map[default_proxy_index]
  if proxy_map.key? country
    proxy_address = proxy_map[country]
  end

  proxy_address = "http://customer-#{get_setting(:Username)}:#{get_setting(:Password)}@#{proxy_address}"

  [url, proxy_address]
end